sdk/core/Azure.Core/src/Serialization/SchemaValidator.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// This abstract class allows any available library to be used to generate schemas from .NET types and validate
+    /// objects against schemas.
+    /// </summary>
+    /// <remarks>
+    /// Defining both <see cref="GenerateSchema(Type)"/> and <see cref="Validate(object, Type, string)"/> is required. If you
+    /// do not wish to validate, then evaluate all schemas as valid.
+    /// </remarks>
+    public abstract class SchemaValidator
```

Comment:
Should we consider making this `SchemaValidator<TSchema>`  to allow the type of schema definition to be controlled by the implementation?    I'm thinking something like:

```csharp
public abstract class SchemaValidator<T>
{
    public abstract void Validate(object data, Type dataType, T schemaDefinition);
    public abstract T GenerateSchema(Type dataType);
}
```

**Example use:**
```csharp
public class SchemaRegistryJsonSchemaValidator : SchemaValidator<string>
{
    public override Validate(object data, string schemaDefinition) => throw new NotImplementedException();
    public override string GenerateSchema(type dataType) => throw new NotImplementedException(); 
}
```

Reply:
That looks really nice, and would probably be a lot more general, my only concern is when we're integrating with the client we need it to be in string form. So if the user decides to return anything that doesn't have a good ToString implementation then it might end up being more confusing?

For example when I was testing out one of the third-party libraries, they have a schema type but you have to do something like schema.Properties.ActualSchema.ToString() to get it in string form. Just doing schema.ToString() gives you the default object.tostring result

Maybe we could have the `SchemaValidator<TSchema>` version in Azure.Core and the `SchemaRegistrySerializer` can require a `SchemaValidator<string>`?

Reply:
The parameter signature in Schema Registry would still be `SchemaValidator<string>`, so implementations would have to return `string` to be valid; this just offers some flexibility for use outside the Schema Registry or JSON cases.

Reply:
Updating to this.